Index: tr181-dns-v2.7.1/scripts/tr181-dns.sh
===================================================================
--- tr181-dns-v2.7.1.orig/scripts/tr181-dns.sh
+++ tr181-dns-v2.7.1/scripts/tr181-dns.sh
@@ -1,21 +1,29 @@
 #!/bin/sh
-[ -f /etc/environment ] && source /etc/environment
-ulimit -c ${ULIMIT_CONFIGURATION:-0}
+
+. /usr/lib/amx/scripts/amx_init_functions.sh
+
 name="tr181-dns"
+datamodel_root="DNS"
 
 case $1 in
-    start|boot)
-	${name} -D
+    boot)
+        process_boot ${name} -D
         ;;
-    stop|shutdown)
-        if [ -f /var/run/${name}.pid ]; then
-            kill "$(cat /var/run/${name}.pid)"
-        else
-            killall ${name}
-        fi
+    start)
+        process_start ${name} -D
+        ;;
+    stop)
+        process_stop ${name}
+        ;;
+    shutdown)
+        process_shutdown ${name}
+        ;;
+    restart)
+        $0 stop
+        $0 start
         ;;
     debuginfo)
-        ubus-cli "DNS.?"
+        process_debug_info ${datamodel_root}
         if [ -f /etc/unbound/unbound.conf ]; then
             printf "\ncat /etc/unbound/unbound.conf\n"
             cat /etc/unbound/unbound.conf
@@ -27,10 +35,6 @@ case $1 in
             cat "$f"
         done
         ;;
-    restart)
-        $0 stop
-        $0 start
-        ;;
     *)
         echo "Usage : $0 [start|boot|stop|shutdown|debuginfo|restart]"
         ;;
