Index: tr181-ethernet-v1.20.1/scripts/ethernet-manager.sh
===================================================================
--- tr181-ethernet-v1.20.1.orig/scripts/ethernet-manager.sh
+++ tr181-ethernet-v1.20.1/scripts/ethernet-manager.sh
@@ -1,29 +1,31 @@
 #!/bin/sh
-[ -f /etc/environment ] && source /etc/environment
-ulimit -c ${ULIMIT_CONFIGURATION:-0}
+
+. /usr/lib/amx/scripts/amx_init_functions.sh
+
 name="ethernet-manager"
+datamodel_root="Ethernet"
 
 case $1 in
-    start|boot)
-        source /etc/environment
-        ${name} -D
-        ;;
-    stop|shutdown)
-        if [ -f /var/run/${name}.pid ]; then
-            kill `cat /var/run/${name}.pid`
-        else
-            killall ${name}
-	fi
+    boot)
+        process_boot ${name} -D
         ;;
-    debuginfo)
-        ubus-cli "Ethernet.?"
+    start)
+        process_start ${name} -D
+        ;;
+    stop)
+        process_stop ${name}
+        ;;
+    shutdown)
+        process_shutdown ${name}
         ;;
     restart)
         $0 stop
         $0 start
         ;;
+    debuginfo)
+        process_debug_info ${datamodel_root}
+        ;;
     *)
         echo "Usage : $0 [start|boot|stop|shutdown|debuginfo|restart]"
         ;;
 esac
-
