Index: tr181-led-v0.4.7/scripts/tr181-led.sh
===================================================================
--- tr181-led-v0.4.7.orig/scripts/tr181-led.sh
+++ tr181-led-v0.4.7/scripts/tr181-led.sh
@@ -1,28 +1,30 @@
 #!/bin/sh
 
-[ -f /etc/environment ] && source /etc/environment
-ulimit -c ${ULIMIT_CONFIGURATION:-0}
+. /usr/lib/amx/scripts/amx_init_functions.sh
+
 name="tr181-led"
+datamodel_root="LEDs"
 
 case $1 in
-    start|boot)
-        ${name} -D
+    boot)
+        process_boot ${name} -D
         ;;
-    stop|shutdown)
-        if [ -f /var/run/${name}.pid ]; then
-            kill `cat /var/run/${name}.pid`
-        else
-            killall ${name}
-        fi
+    start)
+        process_start ${name} -D
         ;;
-    debuginfo)
-        ubus-cli "LEDs.?"
+    stop)
+        process_stop ${name}
+        ;;
+    shutdown)
+        process_shutdown ${name}
         ;;
     restart)
         $0 stop
-        sleep 1
         $0 start
         ;;
+    debuginfo)
+        process_debug_info ${datamodel_root}
+        ;;
     *)
         echo "Usage : $0 [start|boot|stop|shutdown|debuginfo|restart]"
         ;;
