Index: tr181-security-v0.3.0/scripts/tr181-security.sh
===================================================================
--- tr181-security-v0.3.0.orig/scripts/tr181-security.sh
+++ tr181-security-v0.3.0/scripts/tr181-security.sh
@@ -1,23 +1,30 @@
 #!/bin/sh
 
+. /usr/lib/amx/scripts/amx_init_functions.sh
+
 name="tr181-security"
+datamodel_root="Security"
 
 case $1 in
-    start|boot)
-        ${name} -D
+    boot)
+        process_boot ${name} -D
         ;;
-    stop|shutdown)
-        if [ -f "/var/run/${name}.pid" ]; then
-            kill $(cat "/var/run/${name}.pid")
-        fi
+    start)
+        process_start ${name} -D
         ;;
-    debuginfo)
-        ubus-cli "Security.?"
+    stop)
+        process_stop ${name}
+        ;;
+    shutdown)
+        process_shutdown ${name}
         ;;
-    restart|reboot)
+    restart)
         $0 stop
         $0 start
         ;;
+    debuginfo)
+        process_debug_info ${datamodel_root}
+        ;;
     *)
         echo "Usage : $0 [start|boot|stop|shutdown|debuginfo|restart]"
         ;;
