--- a/scripts/tr181-xpon.sh
+++ b/scripts/tr181-xpon.sh
@@ -1,5 +1,6 @@
 #!/bin/sh
 
+. /usr/lib/amx/scripts/amx_init_functions.sh
 [ -f /etc/environment ] && . /etc/environment
 # shellcheck disable=SC3045
 ulimit -c "${ULIMIT_CONFIGURATION:-0}"
@@ -20,11 +21,7 @@ case $1 in
         ${name} -D
         ;;
     stop|shutdown)
-        if [ -f /var/run/${name}.pid ]; then
-            kill "$(cat /var/run/${name}.pid)"
-        else
-            killall ${name}
-        fi
+        process_stop_shutdown ${name} 5
         ubus call ProxyManager unregister "{'proxy' : '$proxypath','real' : '$realpath'}"
         ;;
     debuginfo|d)
