Index: tr181-deviceinfo-v2.33.11/makefile
===================================================================
--- tr181-deviceinfo-v2.33.11.orig/makefile
+++ tr181-deviceinfo-v2.33.11/makefile
@@ -98,7 +98,6 @@ endif
 	$(INSTALL) -D -p -m 0755 scripts/logrotate_mv.sh $(DEST)/usr/lib/debuginfo/logrotate_mv.sh
 	$(INSTALL) -D -p -m 0644 output/$(MACHINE)/mod-lr-logrotate.so $(DEST)/usr/lib/amx/modules/mod-lr-logrotate.so
 	$(INSTALL) -D -p -m 0644 output/$(MACHINE)/mod-syslog.so $(DEST)/usr/lib/amx/modules/mod-syslog.so
-	$(INSTALL) -D -p -m 0755 scripts/amx_init_functions.sh $(DEST)/usr/lib/amx/scripts/amx_init_functions.sh
 
 package: all
 	$(INSTALL) -D -p -m 0644 output/$(MACHINE)/$(COMPONENT).so $(PKGDIR)/usr/lib/amx/$(COMPONENT)/$(COMPONENT).so
@@ -167,7 +166,6 @@ endif
 	$(INSTALL) -D -p -m 0755 scripts/logrotate_mv.sh $(PKGDIR)/usr/lib/debuginfo/logrotate_mv.sh
 	$(INSTALL) -D -p -m 0644 output/$(MACHINE)/mod-lr-logrotate.so $(PKGDIR)/usr/lib/amx/modules/mod-lr-logrotate.so
 	$(INSTALL) -D -p -m 0644 output/$(MACHINE)/mod-syslog.so $(PKGDIR)/usr/lib/amx/modules/mod-syslog.so
-	$(INSTALL) -D -p -m 0755 scripts/amx_init_functions.sh $(PKGDIR)/usr/lib/amx/scripts/amx_init_functions.sh
 	cd $(PKGDIR) && $(TAR) -czvf ../$(COMPONENT)-$(VERSION).tar.gz .
 	cp $(PKGDIR)../$(COMPONENT)-$(VERSION).tar.gz .
 	make -C packages
@@ -196,4 +194,4 @@ test:
 	$(MAKE) -C test run
 	$(MAKE) -C test coverage
 
-.PHONY: all clean changelog install package doc test
\ No newline at end of file
+.PHONY: all clean changelog install package doc test
Index: tr181-deviceinfo-v2.33.11/scripts/amx_init_functions.sh
===================================================================
--- tr181-deviceinfo-v2.33.11.orig/scripts/amx_init_functions.sh
+++ /dev/null
@@ -1,29 +0,0 @@
-#!/bin/sh
-
-process_stop_shutdown(){
-	local name="$1"
-	local pid=$(cat "/var/run/${name}.pid")
-	local timeout_sec="$2"
-	local wait=0
-
-    if [ -z "$pid" ]; then
-        logger "Process $name - $pid not found. Nothing to stop..."
-    else
-        logger "Sending SIGTERM to process $name - $pid..."
-        kill "$pid"
-
-        # Wait for the process to terminate (with timeout)
-        while [ $wait -lt $timeout_sec ] && ps -p "$pid" > /dev/null; do
-            sleep 1
-            wait=$((wait+1))
-        done
-
-        # Check if the process is still running
-        if ps -p "$pid" > /dev/null; then
-            logger "Process $name - $pid did not terminate within $timeout_sec seconds. Sending SIGKILL..."
-            kill -9 "$pid"
-        else
-            logger "Process $name - $pid stopped successfully."
-        fi
-    fi
-}
\ No newline at end of file
Index: tr181-deviceinfo-v2.33.11/scripts/deviceinfo-manager.sh
===================================================================
--- tr181-deviceinfo-v2.33.11.orig/scripts/deviceinfo-manager.sh
+++ tr181-deviceinfo-v2.33.11/scripts/deviceinfo-manager.sh
@@ -1,24 +1,31 @@
 #!/bin/sh
-source /usr/lib/amx/scripts/amx_init_functions.sh
-[ -f /etc/environment ] && source /etc/environment
-ulimit -c ${ULIMIT_CONFIGURATION:-0}
+
+. /usr/lib/amx/scripts/amx_init_functions.sh
+
 name="deviceinfo-manager"
+datamodel_root="DeviceInfo"
 
 case $1 in
-    start|boot)
-        $name -D
+    boot)
+        process_boot ${name} -D
         ;;
-    stop|shutdown)
-        process_stop_shutdown ${name} 5
+    start)
+        process_start ${name} -D
         ;;
-    debuginfo)
-        ba-cli "DeviceInfo.?"
+    stop)
+        process_stop ${name}
+        ;;
+    shutdown)
+        process_shutdown ${name}
         ;;
     restart)
         $0 stop
         $0 start
         ;;
+    debuginfo)
+        process_debug_info ${datamodel_root}
+        ;;
     *)
-        echo "Usage : $0 [start|boot|stop|debuginfo|log]"
+        echo "Usage : $0 [start|boot|stop|shutdown|debuginfo|restart]"
         ;;
 esac
Index: tr181-deviceinfo-v2.33.11/scripts/deviceinfo-system.sh
===================================================================
--- tr181-deviceinfo-v2.33.11.orig/scripts/deviceinfo-system.sh
+++ tr181-deviceinfo-v2.33.11/scripts/deviceinfo-system.sh
@@ -1,24 +1,31 @@
 #!/bin/sh
-source /usr/lib/amx/scripts/amx_init_functions.sh
-[ -f /etc/environment ] && source /etc/environment
-ulimit -c ${ULIMIT_CONFIGURATION:-0}
+
+. /usr/lib/amx/scripts/amx_init_functions.sh
+
 name="deviceinfo-system"
+datamodel_root="System"
 
 case $1 in
-    start|boot)
-        $name -D
+    boot)
+        process_boot ${name} -D
         ;;
-    stop|shutdown)
-        process_stop_shutdown ${name} 5
+    start)
+        process_start ${name} -D
         ;;
-    debuginfo)
-        ba-cli "System.?"
+    stop)
+        process_stop ${name}
+        ;;
+    shutdown)
+        process_shutdown ${name}
         ;;
     restart)
         $0 stop
         $0 start
         ;;
+    debuginfo)
+        process_debug_info ${datamodel_root}
+        ;;
     *)
-        echo "Usage : $0 [start|boot|stop|debuginfo|log]"
+        echo "Usage : $0 [start|boot|stop|shutdown|debuginfo|restart]"
         ;;
 esac
