Index: tr181-httpaccess-v0.10.3/scripts/tr181-httpaccess.sh
===================================================================
--- tr181-httpaccess-v0.10.3.orig/scripts/tr181-httpaccess.sh
+++ tr181-httpaccess-v0.10.3/scripts/tr181-httpaccess.sh
@@ -1,30 +1,31 @@
 #!/bin/sh
-[ -f /etc/environment ] && source /etc/environment
-ulimit -c ${ULIMIT_CONFIGURATION:-0}
+
+. /usr/lib/amx/scripts/amx_init_functions.sh
+
 name="tr181-httpaccess"
+datamodel_root="UserInterface"
 
 case $1 in
-    start|boot)
-	    ${name} -D
+    boot)
+        process_boot ${name} -D
         ;;
-    stop|shutdown)
-        if [ -f /var/run/${name}.pid ]; then
-            kill `cat /var/run/${name}.pid`
-        else
-            killall ${name}
-        fi
+    start)
+        process_start ${name} -D
         ;;
-    debuginfo)
-	ubus-cli "UserInterface.?"
+    stop)
+        process_stop ${name}
+        ;;
+    shutdown)
+        process_shutdown ${name}
         ;;
     restart)
         $0 stop
         $0 start
         ;;
-    log)
-	echo "TODO log httpaccess manager"
-	;;
+    debuginfo)
+        process_debug_info ${datamodel_root}
+        ;;
     *)
-        echo "Usage : $0 [start|boot|stop|debuginfo|log]"
+        echo "Usage : $0 [start|boot|stop|shutdown|debuginfo|restart]"
         ;;
 esac
