Index: packet-interception-v0.0.14/scripts/packet-interception.sh
===================================================================
--- packet-interception-v0.0.14.orig/scripts/packet-interception.sh
+++ packet-interception-v0.0.14/scripts/packet-interception.sh
@@ -1,26 +1,33 @@
 #!/bin/sh
 
+source /usr/lib/amx/scripts/amx_init_functions.sh
 FW_DEFAULTS=/etc/amx/packet-interception/packet-interception.defaults
 FW6_DEFAULTS=/etc/amx/packet-interception/packet-interception_ipv6.defaults
 
 name="packet-interception"
+datamodel_root="PacketInterception"
 
 case $1 in
-    start|boot)
-        ${name} -D /etc/amx/packet-interception/packet-interception.odl
+    boot)
+        process_boot ${name} -D
         ;;
-    stop|shutdown)
-        if [ -f /var/run/${name}.pid ]; then
-            kill `cat /var/run/${name}.pid`
-        else
-            killall ${name}
-        fi
+    start)
+        process_start ${name} -D
+        ;;
+    stop)
+        process_stop ${name}
+        ;;
+    shutdown)
+        process_shutdown ${name}
         ;;
     restart)
         $0 stop
         $0 start
         ;;
+    debuginfo)
+        process_debug_info ${datamodel_root}
+        ;;
     *)
-        echo "Usage : $0 [start|boot|stop|log]"
+        echo "Usage : $0 [start|boot|stop|shutdown|debuginfo|restart]"
         ;;
 esac
